packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart

```diff
@@ -1636,4 +1635,59 @@ void main() {
     controller2.jumpTo(1.0);
     expect(loggedSizes, <double>[1.0].map((double v) => closeTo(v, precisionErrorTolerance)));
   });
+
+  testWidgets('DraggableScrollableSheet controller can be changed while animating', (WidgetTester tester) async {
+    final DraggableScrollableController controller1 = DraggableScrollableController();
+    final DraggableScrollableController controller2 = DraggableScrollableController();
+
+    DraggableScrollableController controller = controller1;
+    await tester.pumpWidget(MaterialApp(
+      home: StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) => Scaffold(
+          body: DraggableScrollableSheet(
+            initialChildSize: 0.25,
+            snap: true,
+            snapSizes: const <double>[0.25, 0.5, 1.0],
+            controller: controller,
+            builder: (BuildContext context, ScrollController scrollController) {
+              return ListView(
+                controller: scrollController,
+                children: <Widget>[
+                  ElevatedButton(
+                    onPressed: () => setState(() {
+                      controller = controller2;
+                    }),
+                    child: const Text('Switch controller'),
+                  ),
+                  Container(
+                    height: 10000,
+                    color: Colors.blue,
+                  ),
+                ],
+              );
+            },
+          ),
+        ),
+      ),
+    ));
+    expect(controller1.isAttached, true);
+    expect(controller2.isAttached, false);
+
+
+    controller1.animateTo(0.5, curve: Curves.linear, duration: const Duration(milliseconds: 200));
```

Comment:
This is the key difference between the test above this one - in the test above, `jumpTo` is called instead of `animateTo`. `jumpTo` doesn't have any ongoing state/actions between the `didUpdateWidget`, whereas `animateTo` does.